onnxscript/ir/_linked_list.py

```diff
@@ -133,9 +125,12 @@ def __reversed__(self) -> Iterator[TLinkable]:
             box = box.prev
 
     def __len__(self) -> int:
+        assert self._length == len(
+            self._values_to_boxes
+        ), "Bug in the implementation: length mismatch"
         return self._length
 
-    def __getitem__(self, index: int) -> TLinkable:
+    def __getitem__(self, index: int) -> T:
```

Comment:
I wonder if we should forbid this for now, rather than create the impression of a constant-time access ... because we don't expect users to use this kind of access, do we?

Reply:
Since the node is a sequence, I find it convenient to find a node by index, for example when I was exploring the model in the console. I think alternatively we can call out the runtime complexity in the docstring?

Reply:
Documented. I will update the documentation for graph itself in a separate PR.